src/pages/gen2/build-a-backend/data/customize-authz/public-data-access/index.mdx

```diff
@@ -40,3 +60,49 @@ const schema = a.schema({
     .authorization([a.allow.public('iam')]),
 });
 ```
+
+In your application, you can perform CRUD operations against the model using `client.models.<ModelName>` with the `iam` auth mode.
+
+<Callout info>
+In the Amplify Library's client configuration file (`amplifyconfiguration.json`) set `allowGuestAccess` to `true`. This lets the Amplify Library use the unauthenticated role from your Cognito identity pool when your user isn't logged in.
```

Comment:
Now that you mention it, I assumed it was required but maybe not in Gen 2?
I just removed it from my config and `iam` auth mode graphql requests still seem to be signed and authorized correctly whether i'm logged in or not.

Is this because we enable unauthenticated access on the user pool in Gen 2 by default? Maybe this is a Gen 1 requirement? I'll try and find out.

Reply:
yes unauthenticated access is enabled by default for gen2 backends, though I'm not sure there's a key for the config file for the library to be aware. in any event, this file is automatically generated and will be overwritten on the next deployment, which makes manual edits tedious to maintain.